walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -121,6 +184,68 @@ public void everyAttendeeIsConsidered() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void optionalAttendeeIsNeverAvailable() {
+    // Have each mandatory atendee have different events, and a single optional attendee who has an 
+    // all day event. We should see three options because each person has
+    // split the restricted times.
+    //
+    // Events  :       |--A--|     |--B--|
+    //         : |-------------C---------------|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_C)),
+        new Event("Event 3", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optionalAttendeeHasTwoOverlaps() {
+    // Have each mandatory attendee have different events, and a single optional attendee who has a single event
+    // We should see only wo options that work for all three attendees.
```

Comment:
```suggestion
    // We should see only two options that work for all three attendees.
```

Btw i like these diagrams, super helpful